### PR TITLE
created User model with validations and seads

### PR DESCRIPTION
**Pull request by Andrii-Kozak**

### CODE REVIEW

**Comment by chu4** [comment]
add a method for full_name

**Comment by pereslop** [comment]
here will be useful to add  tests for model validations, you can add factory-bot and faker gems to create test user instances, user factory should be created

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
+  validates :email, presence: true, format: { with: VALID_EMAIL_REGEX }, uniqueness: true
+
+
+  def full_name
```
Comment (pereslop) [inline app/models/user.rb:9]: please add tests for this method

**Review by pereslop — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 1 above)
Comment (vasya539) [reply app/models/user.rb:9]: validates :first_name, last_name, email, presence: true
validates :email, format: { with: VALID_EMAIL_REGEX }, uniqueness: true

File: db/seeds.rb [hunk 2]
```diff
@@ -1,7 +1,4 @@
...
-# The data can then be loaded with the rails db:seed command (or created alongside the database with db:setup).
-#
-# Examples:
-#
-#   movies = Movie.create([{ name: 'Star Wars' }, { name: 'Lord of the Rings' }])
-#   Character.create(name: 'Luke', movie: movies.first)
+# Create a main sample user.
+User.create!(first_name: "Vasia",
+             last_name: "Popkin",
+             email: "example@popkin.org",)
```
Comment (vasya539) [inline db/seeds.rb:4]: не знаю як у вас там по таскам, але є резон налаштути faker з FactoryBot, і зних сідай вже данні

File: spec/factories/users.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+FactoryBot.define do
+  factory :user do
+    first_name { "John" }
```
Comment (chu4) [inline spec/factories/users.rb:3]: user faker for names as well

**Review by chu4 — APPROVED** [review]

**Review by marikrebega — APPROVED** [review]

**Review by matsevich — APPROVED** [review]

File: db/migrate/20200930080214_create_users.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class CreateUsers < ActiveRecord::Migration[6.0]
+  def change
+    create_table :users do |t|
```
Comment (sviatoslav-krupa) [inline db/migrate/20200930080214_create_users.rb:3]: Add indexes and constraints, please

Comment (sviatoslav-krupa) [inline db/migrate/20200930080214_create_users.rb:3]: Add indexes and constraints, please

**Review by sviatoslav-krupa — CHANGES_REQUESTED** [review]

File: db/migrate/20200930080214_create_users.rb (hunk 4 above)
Comment (sviatoslav-krupa) [inline db/migrate/20200930080214_create_users.rb:3]: Add indexes and constraints, please

**Review by sviatoslav-krupa — CHANGES_REQUESTED** [review]

File: db/schema.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+ActiveRecord::Schema.define(version: 2020_09_30_080214) do
+
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "users", force: :cascade do |t|
+    t.string "first_name"
+    t.string "last_name"
+    t.string "email"
+    t.datetime "created_at", precision: 6, null: false
```
Comment (pereslop) [inline db/schema.rb:22]: here enough to add `t.timestamps`, that will create created_at and updated_columns

File: db/migrate/20200930080214_create_users.rb (hunk 4 above)
Comment (Andrii-Kozak) [reply db/migrate/20200930080214_create_users.rb:3]: @sviatoslav-krupa fixed index by email, is it ok?

Comment (sviatoslav-krupa) [reply db/migrate/20200930080214_create_users.rb:3]: okay, but what about constraints?

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i.freeze
+
+  validates :first_name, :last_name, :email, presence: true
+  validates :email, format: { with: VALID_EMAIL_REGEX }, uniqueness: true
```
Comment (serhii-bodnaruk) [inline app/models/user.rb:5]: use the constant built into URI in the standard ruby library
`URI::MailTo::EMAIL_REGEXP`

**Review by serhii-bodnaruk — APPROVED** [review]

**Pull request merged**